File: flow/distiller.js
export function distill(memoryData) {
      if (!Array.isArray(memoryData)) return [];

  return memoryData.filter(item => {
          if (item.keep === true) return true;
          if (item.avoid === true) return false;
          if (item.note && item.note.trim().length > 5) return true;
          return false;
  });
}

File: ruby/distiller.rb
# distiller.rb - Noise Filter
def distill(memory_data)
  return [] unless memory_data.is_a?(Array)
  memory_data.select do |item|
    next true  if item[:keep] == true
    next false if item[:avoid] == true
    item[:note] && !item[:note].strip.empty?
  end
end
